docs/DeveloperGuide.md

```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
+      
+      Use case resumes from step 3.
+
+**Use case: UC13 - Mark a deliverable as completed**
+
+**Precondition(s):**
+* User is in the deliverable mode.
+* Deliverable to mark as completed exists.
+
+**Guarantee(s):**
+* Upon successful command, the OCP will be updated accordingly.
 
-1. User requests to list contacts.
-1. Productiv shows list of contacts.
-1. User edits a specific contact in the list.
-1. Productiv edits contact details.
+**MSS**
+1. User chooses to mark a deliverable as completed.
+2. User enters the command to mark a deliverable as completed into the command box.
+3. Productiv marks the deliverable as completed.
 
     Use case ends.
 
 **Extensions**
-
-* 2a. The list is empty.
-  
-  * Use case ends.
-     
-* 3a. The given index is invalid.
-    
-    * 3a1. Productiv shows an error message.
-    
-      Use case resumes at step 2.
+* 2a. Productiv detects an error in the command.
+    * 2a1. Productiv displays an error message.
+    * 2a2. User enters the command again.
+    * Steps 2a1-2a2 are repeated until the command entered is correct.
       
-* 4a. The given input is invalid.
-    
-    * 4a1 Productiv shows an error message.
+      Use case resumes from step 3.
       
-      Use case resumes at step 2.
-
-**Use case: `UC12 - Delete a contact`**
+**Use case: UC14 - Mark a deliverable as on-going**
 
 **Precondition(s):**
-* **`User is in the Contact mode`** 
-* **`Contact to delete exists`**
+* User is in the deliverable mode.
+* Deliverable to mark as on-going exists.
 
-**Guarantee(s):** 
-* **`Deleted contacts will not have its data in Productiv`**
-* **`Deleted contacts cannot be retrieved back`**
+**Guarantee(s):**
+* Upon successful command, the OCP will be updated accordingly.
 
 **MSS**
-
-1. User requests to list contacts.
-1. Productiv shows list of contacts.
-1. User requests to delete a specific contact in the list.
-1. Productiv deletes the contact.
-   User case ends.
+1. User chooses to mark a deliverable as on-going.
+2. User enters the command to mark a deliverable as on-going into the command box.
+3. Productiv marks the deliverable as on-going.
 
     Use case ends.
 
 **Extensions**
-
-* 2a. The list is empty.
-  
-  * Use case ends.
-     
-* 3a. The given index is invalid.
-    
-    * 3a1. Productiv shows an error message.
-    
-      Use case resumes at step 2.
+* 2a. Productiv detects an error in the command.
+    * 2a1. Productiv displays an error message.
+    * 2a2. User enters the command again.
+    * Steps 2a1-2a2 are repeated until the command entered is correct.
       
+      Use case resumes from step 3.
+    
+**Use case: UC15 - View a deliverable**
 
+**Precondition(s):**
+* User is in the deliverable mode.
+* Deliverable to view exists.
 
+**MSS**
+1. User chooses to view a deliverable.
+2. User enters the command to view a deliverable into the command box.
+3. Productiv displays the deliverable in the view box.
```

Comment:
right panel? view panel? view box? im not too sure either, but i used right panel